src/compiler/checker.ts

```diff
@@ -8540,6 +8551,13 @@ namespace ts {
             return undefined;
         }
 
+        function isIife(func: FunctionExpression | MethodDeclaration) {
+            return (func.kind === SyntaxKind.FunctionExpression || func.kind === SyntaxKind.ArrowFunction) &&
+                func.parent.kind === SyntaxKind.ParenthesizedExpression &&

```

Comment:
Done, although skipParenthesizedNodes actually goes down, not up (`expression = expression.expression`), so I just wrote an inline loop to go up (`parent = parent.parent`)